servers/rendering/renderer_rd/shaders/effects/blur_raster.glsl

```diff
@@ -94,13 +96,24 @@ void main() {
 	if (bool(blur.flags & FLAG_HORIZONTAL)) {
 		vec2 pix_size = blur.pixel_size;
 		pix_size *= 0.5; //reading from larger buffer, so use more samples
+
 		vec4 color = texture(source_color, uv_interp + vec2(0.0, 0.0) * pix_size) * 0.174938;
 		GLOW_ADD(vec2(1.0, 0.0), 0.165569);
 		GLOW_ADD(vec2(2.0, 0.0), 0.140367);
 		GLOW_ADD(vec2(3.0, 0.0), 0.106595);
 		GLOW_ADD(vec2(-1.0, 0.0), 0.165569);
 		GLOW_ADD(vec2(-2.0, 0.0), 0.140367);
 		GLOW_ADD(vec2(-3.0, 0.0), 0.106595);
+
+		if (bool(blur.flags & FLAG_GLOW_FIRST_PASS)) {
+			// In the first pass bring back to correct color range else we're applying the wrong threshold
+			// in subsequent passes we can use it as is as we'd just be undoing it right after.
+			color *= blur.luminance_multiplier;
+
+			// Undo tonemap to restore range: https://graphicrants.blogspot.com/2013/12/tone-mapping.html
+			color /= 1.0 - dot(color.rgb, vec3(0.299, 0.587, 0.114));
```

Comment:
The samples were not tonemapped in the first place, so the reverse tonemapping shouldnt be applied.

Reply:
I wasn't sure about this either, I don't why this is being done in the clustered one, I just copied it from there as a possible fix.

Reply:
Ah funky, with this change removed `float luminance = dot(frag_color.rgb, vec3(0.299, 0.587, 0.114));` no longer gives the right result, so I've undone that change too.

This was all to get the color brightness right but as explained further down below, I think that is caused by a different issue.

I do wonder if the dot approach for calculating luminance, instead of simply taking the max or r,g or b, is the better approach. We'll have to talk with @JFonS about that one.

For now I've reverted the change, we can always add a specific PR in to improve this.

Reply:
Okay, removing `color /= 1.0 - dot(color.rgb, vec3(0.299, 0.587, 0.114));` and changing the luminance calculation back to what it was results in the mobile renderer blur to be much brighter then the clustered one. A reverse of the previous situation.

However putting it back in the size of the blur become nearly identical with the same intensity values. The problem however is that the mobile renderer is again starting to max out on the glow buffers because you really need to crank up the strength of the glow to get a nice blur and we're hitting max values in the glow buffers themselves now.

Also both with the clustered and mobile renderer, it seems like blue is nearly filtered out, not surprising seeing the values we're doing the dot product with.

I don't know enough about the proper color calculations needed here but Clayjohns remark that the colors aren't tone mapped is spot on, tonemapping doesn't happen until after the glow buffers have been created and applied in the tonemap shader. I'm feeling that what the clustered renderer is doing is actually wrong...

I'm keeping the tonemapping thing in both shaders for now so they are atleast similar in result, however I think this needs more debate (also posted this in `#rendering`)